plugin/wpephpcompat.php

```diff
@@ -84,6 +84,76 @@ function wpe_phpcompat_autoloader( $class ) {
 	}
 }
 
-spl_autoload_register(  __NAMESPACE__ . '\wpe_phpcompat_autoloader' );
+spl_autoload_register( __NAMESPACE__ . '\wpe_phpcompat_autoloader' );
 
 add_action( 'plugins_loaded', __NAMESPACE__ . '\wpe_phpcompat_loader' );
+
+/**
+ * Remove old options and custom post type
+ *
+ * @return void
+ */
+function clean_legacy_data() {
```

Comment:
Seems like this only needs to happen once, is that correct? Right now we run this anytime the plugin is activated, deactivated or updated. Wondering if we should store an option once this runs once that we can use to short circuit here in the future?

Reply:
Thanks for suggestion this @dkotter 

I've added few changes:

1. Renamed the function to `maybe_migrate_to_wptide` for consistency
2. Added an option `wpephpcompat_is_wptide` to save the state after the cleanup
3. Added conditional check at the beginnign to skip the cleanup if already run before
4. The option is removed during the uninstall. In this case, the cleanup will happen again once the plugin is installed after being uninstalled. But I think this is a good trade off.